src/modules/dashboard/sagas/colony.ts

```diff
@@ -348,14 +355,57 @@ function* colonyTokenBalanceFetch({
       payload: {
         token: {
           address: tokenAddress,
-          balance,
+          balances: {
+            [domainId]: balance,
+          },
         },
         tokenAddress,
         colonyAddress,
       },
     });
-  } catch (error) {
-    return yield putError(ActionTypes.COLONY_TOKEN_BALANCE_FETCH_ERROR, error);
+  } catch (caughtError) {
+    return yield putError(
+      ActionTypes.COLONY_TOKEN_BALANCE_FETCH_ERROR,
+      caughtError,
+    );
+  }
+  return null;
+}
+
+function* colonyTokenBalancesFetch({
+  payload: { colonyAddress, tokenAddress },
+}: Action<ActionTypes.COLONY_TOKEN_BALANCES_FETCH>) {
+  try {
+    const { record: domains } = yield selectAsJS(
+      colonyDomainsSelector,
+      colonyAddress,
+    );
+
+    yield all([
+      // fetch balances for rewards pots and non-rewards pots
+      put(
+        fetchColonyTokenBalance(
+          colonyAddress,
+          tokenAddress,
+          COLONY_TOTAL_BALANCE_DOMAIN_ID,
+        ),
+      ),
+      // fetch balances for root domain
+      put(fetchColonyTokenBalance(colonyAddress, tokenAddress, 1)),
+      // fetch balances for other domains
+      ...domains.map(({ id: domainId }) =>
```

Comment:
This is causing an error when `domains` is undefined (still being loaded). I can reproduce by directly loading `/colony/mycolony/admin`.